content/docs/examples/advanced-gateways/http-proxy/index.md

```diff
@@ -0,0 +1,240 @@
+---
+title: Connect to an external HTTPS proxy
+description: Describes how to configure Istio to let applications use an external HTTPS proxy.
+weight: 60
+keywords: [traffic-management,egress]
+---
+The [Configure an Egress Gateway](/docs/examples/advanced-gateways/egress-gateway/) showed how you can direct traffic to
+external services from your mesh via an Istio edge component called _Egress Gateway_. However, there are cases when you
+must use an external, legacy (non-Istio) HTTPS proxy to access external services. For example, your company may already
+have such a proxy in place and all the applications within the organization may be required to direct their traffic
+through it. This example shows how to enable access to an external HTTPS proxy.
```

Comment:
I would start a new paragraph after `through it.` and then explain why this scenario is different. I'm not sure this is exactly right, but something like this anyway:

```
though it.

This example shows how to enable access to an external HTTPS proxy.
Configuring access via such a proxy is complicated by the fact that Istio does not support
the HTTP protocol expected of proxy clients and therefore needs to be configured to simply
pass through the requests using a TCP tunnel.
```

Reply:
Yes, I think you are getting to heart of what is bothering me about this example. It seems like a lot of work to setup the proxy, test it, configure istio, test again, just for something that is not very complicated. It almost seems like a lot of noise that could be replaced by a paragraph or two in the docs somewhere. That said, if we do want the example, I think we at least have to give a very brief intro that explains why it's special. I agree that it should be short and high level but not quite as high level as you suggest (just needs to be configured "in some other way") :-)

Reply:
@frankbu Note this comment https://github.com/istio/istio/issues/7302#issuecomment-420725490:

> would love to make this a task or an entry in the guide

and it got three likes. So my guess is that people are interested in such a task/example. See the issue itself https://github.com/istio/istio/issues/7302, it seems it caused some pain to some people.

Regarding Istio examples - my view on them is as follows. An example does not necessarily has to present some complex or special solution, it is just a scenario that some people can implement. In this case, the scenario is special, the solution is simple (with the catch of TCP entry instead of HTTP). The people who have to work with an HTTPS proxy, can follow this example and see that it works (or does not work) in their environment. The example demonstrates to both Istio developers and Istio users that this particular setting works in Istio. Ideally, we should automate all our examples  and test them for every PR.

>  I think we at least have to give a very brief intro that explains why it's special.

Let me add a sentence about HTTP CONNECT.

Reply:
Addressed in https://github.com/istio/istio.io/pull/2960/commits/57fdcd843fa927612770fe774ba35579b71cfe4b